core/src/swarm.rs

```diff
@@ -355,6 +317,10 @@ pub trait NetworkBehaviour<TTopology> {
     /// Builds a new `ProtocolsHandler`.
     fn new_handler(&mut self) -> Self::ProtocolsHandler;
 
+    /// Addresses that this behaviour is aware of for this specific peer, and that may allow
+    /// reaching the peer.
+    fn addresses_of_peer(&self, peer_id: &PeerId) -> Vec<Multiaddr>;
```

Comment:
This should likewise be generic over peer IDs.

Reply:
It's not as easy, because then we have to make the `NetworkBehaviour` trait generic over peer IDs as well.

Reply:
Yes, I know.